pkg/kubelet/winstats/winstats.go

```diff
@@ -103,9 +105,24 @@ func (c *StatsClient) WinContainerInfos() (map[string]cadvisorapiv2.ContainerInf
 }
 
 // WinMachineInfo returns a cadvisorapi.MachineInfo with details about the
-// node machine. Analogous to cadvisor MachineInfo method.
+// node machine. Run the powershell command to get the SystemUUID for Windows node
+// in here if it isn't provided by cadvisor.
 func (c *StatsClient) WinMachineInfo() (*cadvisorapi.MachineInfo, error) {
-	return c.client.getMachineInfo()
+	infos, err := c.client.getMachineInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	if infos.SystemUUID == "" {
+		cmd := exec.Command("powershell.exe", "-Command", "(Get-CimInstance -Class Win32_ComputerSystemProduct).UUID")
```

Comment:
@PatrickLang @benmoss does this look okay to you?

Reply:
With the in-tree vSphere cloud provider we had to switch from using `wmic csproduct get UUID` to `wmic bios get serialnumber`, since the former wasn't reflecting the value set by vSphere correctly. I'm not sure if this .NET function is the same as the `wmic` call, but I am getting it would be.

https://github.com/kubernetes/kubernetes/pull/74919

Reply:
@benmoss Thanks for the information! I tried running the two commands on my windows vm and actually they are returning different result:
```
PS C:\windows\system32> (Get-CimInstance -Class Win32_ComputerSystemProduct).UUID
E6C8AC43-582B-3575-4E1F-6DA170888906

PS C:\windows\system32> wmic csproduct get UUID
UUID
E6C8AC43-582B-3575-4E1F-6DA170888906

PS C:\windows\system32> wmic bios get serialnumber
SerialNumber
GoogleCloud-43ACC8E62B5875354E1F6DA170888906
```
It looks like the first one is more closer to what we want in here. I'm also fine with using the `serialnumber` if that makes more sense. Any suggestions?

Reply:
I think looking at this again we want to stick to UUID, I think the cloud providers can override this value if they need to

Reply:
Thanks Ben!

Reply:
It's not that big of deal since we only run this once during kubelet's lifetime, but running `wmic csproduct get UUID` directly probably slightly faster than calling out to `powershell.exec`.

Reply:
Good suggestion! Done.

Reply:
`wmic csproduct get UUID` should be correct with regards to the vsphere provider. The in-tree vSphere cloud provider doesn't rely on `node.Status.NodeInfo.SystemUUID` anymore, and the out-of-tree vSphere cloud provider handles the UUID mapping - https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/util.go#L48